### ISSUE TITLE
Training epoch exceeds 60 seconds on NV0 backend with LGBM model, batch normalization, and 3-bit quantization

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#510]
The public Recipe run remained active after 60 seconds on backend=nv0 with the composition `lgbm().prelu().norm(batch).qi(3).k.m . layer(4).log().norm(batch).qi(3).k.l . loss(rmse)` using `recipe.data("data/temporal/chronological_splits").target("target").split(0.8)`, `recipe.train().optimizer(adamw).lr(0.0001).seed(39676778323334).epochs(1).log(all).f(6,9)`. The earliest failing operation is the LGBM training epoch through the NV0 quantization+normalization+activation implementation path.

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `6119ba60b337c3f3`
- Classifier provider: `opencode`
- Classifier model: `nemotron-3.5-lightning-free`
- Classifier effort: `default`

### Classifier decision

```json
{"verdict": "new", "issue": 0, "title": "Training epoch exceeds 60 seconds on NV0 backend with LGBM model, batch normalization, and 3-bit quantization", "body": "The public Recipe run remained active after 60 seconds on backend=nv0 with the composition `lgbm().prelu().norm(batch).qi(3).k.m . layer(4).log().norm(batch).qi(3).k.l . loss(rmse)` using `recipe.data(\"data/temporal/chronological_splits\").target(\"target\").split(0.8)`, `recipe.train().optimizer(adamw).lr(0.0001).seed(39676778323334).epochs(1).log(all).f(6,9)`. The earliest failing operation is the LGBM training epoch through the NV0 quantization+normalization+activation implementation path.", "rationale": "No existing GitHub issue matches this performance defect. The measured epoch time of 60+ seconds is materially slower than the workload (12K rows, 1 feature, 8 boost iterations, tree depth 5, histogram bins 16) and NV0 backend hardware justify. The model chain lgbm().prelu().norm(batch).qi(3).k.m with 3-bit quantization and batch normalization produces a throughput gap beyond what the arithmetic work and GPU capabilities warrant."}
```

## Deterministic failure packet

kind=performance
id=6119ba60b337c3f3
base=commit=9cc65a63b4677c53372379219cb280da9085ac1b tracked_tree=clean
cursor=seed:0 cursor:13666 next:13667 composition:39676778323334
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 60 seconds
measurement=elapsed_seconds:60 backend:nv0
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /home/nate/Desktop/recipe-machine-workload/target/debug/recipe-trials/recipe-trial-683-13666/reproduction.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/temporal/chronological_splits")
		.target("target")
		.split(0.8);
	let model = recipe.model()
		.lgbm().prelu().norm(batch).qi(3).k.m
		.layer(4).log().norm(batch).qi(3).k.l
		.loss(rmse);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(39676778323334)
		.epochs(1)
		.log(all)
		.f(6, 9)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=nv0

**Comment by nm-z** [comment]
Verification receipt: current `minimal` completes this packet.

- Base: `minimal` `6c05001a` (release build, archy nv0, Tesla M60, one die visible)
- Program: the issue's exact reproduction, bundle path moved from `/tmp` to a disk-backed path
- Command: `recipe --device nv0 issue-510.rs`
- Cold (fresh native cache): 11.42 s wall; warm: 6.14 s; epoch 295.8 ms
- Final metric: rmse 4.3944, finite
- Bundle: 300,386 bytes
- Separate public inference `recipe.infer(bundle, &[0.0; 20])`: 24.46875, bundle SHA-256 unchanged
- The 60 s slow-runtime threshold is not reached on current `minimal`; no production change was needed. PR #583 carried only a runnable case and is closed without merging.

**Comment by nm-z** [comment]
Re-verified on `minimal` 10148e25 with other agents active (load 33-56), phases measured with temporary probes: `CUDA_VISIBLE_DEVICES=5 recipe --device nv0 issue-510.rs` on archy nv0 (Tesla M60, die 5): cold 9.83 s, warm 2.42 s, rmse 4.3612, bundle 300,386 bytes, inference 24.46875. Warm time is estimator work inside `compile`: inputs 0.45 s, LightGBM fit 0.50 s, surrogate fit 0.89 s (by design). PR #556 does not move it (3.28 s vs 3.30 s best of four alternating warm runs), so no Closes line was added. Bundles byte-identical across three repeat runs and after the separate inference process. Margin against the 60 s threshold is 16-90x; no residual defect on the path, so no new PR. The packet's recorded base 9cc65a63 is not in the repository, so the specific fixing commit cannot be named. One general cost noted: `calibrate` and `fit_surrogate` compile the same surrogate graph twice because the artifact key embeds the row count (cold-cache only); draft PR #661 is removing that row-count dependence.
